homeassistant/config.py

```diff
@@ -669,6 +680,116 @@ def async_process_component_config(hass, config, domain):
     return None
 
 
+def check_ha_config_file(config_dir):
+    """Check if Home Assistant configuration file is valid."""
+    from unittest.mock import patch
+
+    all_errors = []
+    all_success = OrderedDict()
+
+    def _pack_error(package, component, config, message):
+        """Handle errors from packages: _log_pkg_error."""
+        message = "Package {} setup failed. Component {} {}".format(
+            package, component, message)
+        domain = 'homeassistant.packages.{}.{}'.format(package, component)
+        pack_config = core_config[CONF_PACKAGES].get(package, config)
+        all_errors.append((message, domain, pack_config))
+
+    def _comp_error(ex, domain, config):
+        """Handle errors from components: async_log_exception."""
+        all_errors.append(
+            (_format_config_error(ex, domain, config), domain, config))
+
+    # Load configuration.yaml
+    try:
+        config_path = find_config_file(config_dir)
+        if not config_path:
+            return {}, [("File configuration.yaml not found.", None, None)]
+        config = load_yaml_config_file(config_path)
+    except HomeAssistantError as err:
+        return {}, [(str(err), None, None)]
```

Comment:
Let's not return tuples, they are confusing and also mean that any code depending on this function will need to be updated if we want to add another parameter. Please consider using a namedtuple or an attr class.

Reply:
I like the idea behind namedtuples. The current double tuple approach can be confusing, especially in the unit tests